src/clients/HttpTunneler.cc

```diff
@@ -154,7 +154,7 @@ Http::Tunneler::writeRequest()
     MemBuf mb;
 
     try {
-        request->masterXaction->generatingConnect = true;
+        request->generatingConnect = true;
```

Comment:
AFAICT, the `request` here is the request _received_ from the client or equivalent. Commit 090f1d3 message agrees with that conclusion.

We should not be modifying the received request when creating a new request to be sent. To the extent possible, the received request should be treated as (and will eventually become) constant in this context.

There are many places in the current code where HttpRequest is abused to store unrelated information, but I see no reason to add one more such place. The existing MasterXaction-based code does not have this problem.

Reply:
> AFAICT, the `request` here is the request _received_ from the client or equivalent.

The "or equivalent" is important. That means;
 a) there may not be a client, or
 b) the request may be completely different from the one the client sent, or
 c) the client sent it wrapped in nested protocols that we secretly unwrapped without its consent, or
 d) this HttpRequest is just our (HttpStateData's) temporary representation of one of the above.

Reply:
Regarding your const'ness statement. The HttpRequest is not const today, it cannot be const so long as the HTTP request is being processed. This code is part of the HTTP processing.

Please stop referring to "commit https://github.com/squid-cache/squid/commit/090f1d3c3c4e1aa9ab96b84787907967313ccfcf" - the deep technical reason documented there for rejecting this (case 1.2) approach is "It feels strange to consult such an old/"disconnected" message". Time between the parent HttpRequest starting and this new tunnel request being generated is not a valid consideration.